### PR TITLE
Add truncating `LazyList` DOM implementation

### PR DESCRIPTION
**Pull request by veyndan**
The motivation for this was to get rid of `ColumnProvider` in Emoji Search. The `ColumnProvider` abstraction was making it difficult to add `LazyList` specific types as parameters.

Note that this doesn't properly implement `LazyList` on the web, as no items are loaded on scroll. Also, most of the widget properties haven't been implemented.

### CODE REVIEW

File: redwood-lazylayout-dom/build.gradle [hunk 1]
```diff
@@ -0,0 +1,18 @@
+apply plugin: 'org.jetbrains.kotlin.js'
+
+redwoodBuild {
+}
```
Comment (veyndan) [inline redwood-lazylayout-dom/build.gradle:4]: Deliberately not `publishing()` this, as the current implementation doesn't properly implement lazy loading in the DOM. I wanted to make sure we don't accidentally publish a broken version of this. We should enable publishing once #1048 has been completed.

**Review by JakeWharton — APPROVED** [review]
Is this actually "truncating" as described? I didn't see where it happens.

**Comment by veyndan** [comment]
It does, in the sense that an initial N number of items are sent over (as defined in `LazyList.kt`). We never load more, as the host never gives updates to the guest about the current view port, so we only ever display those initial N items.

**Comment by JakeWharton** [comment]
Ah, makes sense.

**Pull request merged**